doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -0,0 +1,90 @@
+
+.. _sampling-tdr:
+
+Transformed Density Rejection (TDR)
+===================================
+
+.. currentmodule:: scipy.stats
+
+* Required: T-concave PDF, dPDF
+* Optional: mode, center
+* Speed:
+
+  * Set-up: slow
+  * Sampling: fast 
+
+
+TDR is an acceptance/rejection method that uses the concavity of a transformed
+density to construct hat function and squeezes automatically. Such PDFs are
+called T-concave. Currently the following transformations are implemented:
+
+.. math::
+
+    c = 0.: T(x) &= \log(x)\\
+    c = -0.5: T(x) &= \frac{1}{\sqrt{x}} \text{ (Default)}
+
+In addition to the PDF, it also requires the derivative of the PDF w.r.t x
+(i.e. the variate). These functions must be present as methods of a python
+object which can then be passed to the generators to instantiate their object.
+
+Three variants of this method are available:
+
+* GW: squeezes between construction points.
+* PS: squeezes proportional to hat function. (Default)
+* IA: same as variant PS but uses a composition method with
+  "immediate acceptance" in the region below the squeeze.
+
+This can be changed by passing a ``variant`` parameter.
+
+An example of using this method is shown below:
+
+    >>> from scipy.stats import TransformedDensityRejection
+    >>> 
+    >>> class Distribution:
+    ...     def pdf(self, x):
+    ...         if abs(x) <= 1:
+    ...             return 1 - x*x
+    ...         return 0
+    ...     def dpdf(self, x):
+    ...         if abs(x) <= 1:
+    ...             return -2*x
+    ...         return 0
+    ... 
+    >>> dist = Distribution()
+    >>> 
+    >>> urng = np.random.default_rng(0xbb51a838c8a087854ad3643b2b268d43)
+    >>> rng = TransformedDensityRejection(dist, seed=urng)
+    >>> rng.rvs()
+    0.39606414669189344
+
+It is also possible to evaluate the inverse of the CDF of the hat distribution
+directly using the ``ppf_hat`` method.
+
+    >>> rng.ppf_hat(0.5)
+    -5.7844272705533106e-05
+    >>> u = np.linspace(0, 1, num=10)
+    >>> rng.ppf_hat(u)
+    array([       -inf, -0.58619335, -0.39060804, -0.22634331, -0.07436531,
+            0.07424874,  0.22622123,  0.39047127,  0.58601716,         inf])
```

Comment:
is it possible to compare this to the true values of the ppf? (not sure if there is a closed form). alternatively, one could compute the approximate ppf value for 0.5 only using Newton's method or sth like that and compare it with ppf_hat(0.5)

Reply:
> is it possible to compare this to the true values of the ppf?

I also tried to compute this. But I don't think there is a closed-form PPF. We can use newton's method to solve for different values but I think that will be too complex for the tutorial. Instead, I can use a different distribution whose PPF is available in the closed-form. What do you think about that?

Reply:
I have used the `norm` distribution now and compared the value of `ppf_hat` with `norm,ppf`.

Reply:
that looks good